.github/workflows/test.yml

```diff
@@ -205,6 +206,86 @@ jobs:
           file: ./coverage.txt
         if: "env.GIT_DIFF != ''"
 
+  test-race-1:
```

Comment:
This seems a bit overkill, I understand it's because some tests fail due to data races. If it can, these tests should be fixed and the entire repo should be run with `-race`

Reply:
I think this is not the scope of this task. 
I agree that we should be able to test everything with race detector.

PS: Do we have an issue to enable detector everywhere?

Reply:
> This seems a bit overkill, I understand it's because some tests fail due to data races. If it can, these tests should be fixed and the entire repo should be run with -race

Running tests with `-race` takes longer than running them without. That's why I split the test runs into multiple jobs. And the tests are not the problem here. Please clone https://github.com/cosmos/cosmos-sdk/pull/7300's branch, and run the tests yourself with `-race`:

* `go test -mod=readonly -race -timeout 15m -coverprofile=coverage.txt -covermode=atomic -tags='ledger test_ledger_mock' ./...` 

There are race conditions that arise in tendermint too.

Reply:
> PS: Do we have an issue to enable detector everywhere?

Not yet but we definitely should!